File: app/src/main/java/com/roy/turbo/launcher/PagedViewWidgetImageView.java
package com.roy.turbo.launcher;

import android.content.Context;
import android.graphics.Canvas;
import android.util.AttributeSet;
import android.widget.ImageView;

//done 2023.03.29
class PagedViewWidgetImageView extends ImageView {
    public boolean mAllowRequestLayout = true;

    public PagedViewWidgetImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void requestLayout() {
        if (mAllowRequestLayout) {
            super.requestLayout();
        }
    }

    @Override
    protected void onDraw(Canvas canvas) {
        canvas.save();
        canvas.clipRect(getScrollX() + getPaddingLeft(),
                getScrollY() + getPaddingTop(),
                getScrollX() + getRight() - getLeft() - getPaddingRight(),
                getScrollY() + getBottom() - getTop() - getPaddingBottom());

        super.onDraw(canvas);
        canvas.restore();

    }
}

File: app/src/main/java/com/roy/turbo/launcher/PagedViewWidgetImageView.kt
package com.roy.turbo.launcher

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet
import android.widget.ImageView

//done 2023.03.29
internal class PagedViewWidgetImageView(context: Context?, attrs: AttributeSet?) :
    ImageView(context, attrs) {
    @JvmField
    var mAllowRequestLayout = true
    override fun requestLayout() {
        if (mAllowRequestLayout) {
            super.requestLayout()
        }
    }

    override fun onDraw(canvas: Canvas) {
        canvas.save()
        canvas.clipRect(
            /* left = */ scrollX + paddingLeft,
            /* top = */ scrollY + paddingTop,
            /* right = */ scrollX + right - left - paddingRight,
            /* bottom = */ scrollY + bottom - top - paddingBottom
        )
        super.onDraw(canvas)
        canvas.restore()
    }
}
